__tests__/event-spec.js

```diff
@@ -93,13 +93,267 @@ describe('Event', () => {
+          method: 'POST',
+          body: {
+            id: undefined,
+            object: 'event',
+            account_id: undefined,
+            calendar_id: undefined,
+            message_id: undefined,
+            busy: undefined,
+            title: undefined,
+            description: undefined,
+            owner: undefined,
+            location: undefined,
+            when: {
+              start_time: 1409594400,
+              end_time: 1409598000,
+            },
+            participants: [],
+            read_only: undefined,
+            status: undefined,
+          },
+          qs: {},
+          path: '/events',
+        });
+        done();
+      });
+    });
+
+    test('should create event with date when the event param `when` is updated with date', done => {
+      testContext.event.when = { date: '1912-06-23' };
+      testContext.event.save().then(() => {
+        expect(testContext.connection.request).toHaveBeenCalledWith({
+          method: 'POST',
+          body: {
+            id: undefined,
+            object: 'event',
+            account_id: undefined,
+            calendar_id: undefined,
+            message_id: undefined,
+            busy: undefined,
+            title: undefined,
+            description: undefined,
+            owner: undefined,
+            location: undefined,
+            when: {
+              date: '1912-06-23',
+            },
+            participants: [],
+            read_only: undefined,
+            status: undefined,
+          },
+          qs: {},
+          path: '/events',
+        });
+        done();
+      });
+    });
+
+    test('should create event with start_date and end_date when the event param `when` is updated with start_date and end_date', done => {
+      testContext.event.when = { start_date: '1815-12-10', end_date: '1852-11-27' };
+      testContext.event.save().then(() => {
+        expect(testContext.connection.request).toHaveBeenCalledWith({
+          method: 'POST',
+          body: {
+            id: undefined,
+            object: 'event',
+            account_id: undefined,
+            calendar_id: undefined,
+            message_id: undefined,
+            busy: undefined,
+            title: undefined,
+            description: undefined,
+            owner: undefined,
+            location: undefined,
+            when: {
+              start_date: '1815-12-10',
+              end_date: '1852-11-27',
+            },
+            participants: [],
+            read_only: undefined,
+            status: undefined,
+          },
+          qs: {},
+          path: '/events',
+        });
+        done();
+      });
+    });
+
     describe('when the request succeeds', () => {
       beforeEach(() => {
         testContext.connection.request = jest.fn(() => {
           const eventJSON = {
             id: 'id-1234',
             title: 'test event',
-            when: {},
+            start: 1409594400,
+            end: 1409594400,
```

Comment:
our API doesn't return start & end, right? if not, let's just include `when` in this example JSON response

Reply:
you're right! i'll remove start and end